.github/workflows/tests.yml

```diff
@@ -19,22 +19,52 @@ jobs:
 
     steps:
     - uses: actions/checkout@v1
+
     - name: Set up Python 3.6
       uses: actions/setup-python@v1
       with:
         python-version: 3.6
+
+    - name: Pip Cache
+      uses: actions/cache@v1
+      id: pip-cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('Pipfile.lock') }}
+        restore-keys: ${{ runner.os }}-pip-
+
+    - name: Pipenv cache
+      uses: actions/cache@v1
+      id: pipenv-cache
+      with:
+        path: ~/.cache/pipenv
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('Pipfile.lock') }}
+        restore-keys: ${{ runner.os }}-pipenv-
+
+    - name: Virtualenv cache
+      uses: actions/cache@v1
+      id: virtualenv-cache
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-virtualenvs-${{ hashFiles('Pipfile.lock') }}
+        restore-keys: ${{ runner.os }}-virtualenvs-
+
+    - name: Install pipenv
+      uses: dschep/install-pipenv-action@v1
+
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip pipenv
         pipenv install --dev
         cp env.json{.sample,} || /bin/true
+
     - name: Lint with flake8
       run: |
         pip install flake8
```

Comment:
Isn't flak8 part of the dev dependencies?

Reply:
Yes, it really should. I made a [commit to do that](https://github.com/Karspexet/Karspexet/pull/238/commits/7d18a46bfcffc511618c164e644bd3100d20a969) last night, but apparently I never pushed it.

Reply:
I also failed utterly when writing that commit, so I will make it work now instead.